packages/plugins/documentation/server/services/documentation.js

```diff
@@ -2,49 +2,69 @@
 
 const path = require('path');
 const fs = require('fs-extra');
-const _ = require('lodash');
+const { cloneDeep } = require('lodash/fp');
 const { getAbsoluteServerUrl } = require('@strapi/utils');
-
 const { builApiEndpointPath, buildComponentSchema } = require('./helpers');
 
 const defaultOpenApiComponents = require('./utils/default-openapi-components');
+const { getPluginsThatNeedDocumentation } = require('./utils/get-plugins-that-need-documentation');
+
+/**
+ * @description
+ * Mutates the current state of the OpenAPI object and returns a new immutable object
+ *
+ * @param {object} currentState The immutable state of the OpenAPI object
+ * @param {function} mutateStateCallback A callback function that can mutate a copy of the immutable state
+ *
+ * @returns {object}
+ * The mutated copy as the new immutable state
+ */
+const mutateDocumentation = (currentState, mutateStateCallback) => {
+  // Create a copy of the current state that is mutable
+  const draftState = cloneDeep(currentState);
+  // Pass the draft to the callback for mutation
+  mutateStateCallback(draftState);
+  // Return the mutated state as a new immutable state
+  return Object.freeze(draftState);
+};
 
 module.exports = ({ strapi }) => {
   const config = strapi.config.get('plugin.documentation');
-  const registeredDocs = [];
+  const pluginsThatNeedDocumentation = getPluginsThatNeedDocumentation(config);
+  const overrideService = strapi.plugin('documentation').service('override');
 
   return {
-    registerDoc(doc, pluginOrigin) {
-      const plugins = this.getPluginsThatNeedDocumentation();
-      let registeredDoc = doc;
-
-      if (pluginOrigin) {
-        if (!plugins.includes(pluginOrigin)) {
-          return strapi.log.info(
-            `@strapi/documentation will not use the override provided by ${pluginOrigin} since the plugin was not specified in the x-strapi-config.plugins array`
-          );
-        }
-      } else {
-        strapi.log.warn(
-          '@strapi/documentation received an override that did not specify its origin, this could cause unexpected schema generation'
-        );
-      }
-
-      // parseYaml
-      if (typeof doc === 'string') {
-        registeredDoc = require('yaml').parse(registeredDoc);
-      }
-      // receive an object we can register it directly
-      registeredDocs.push(registeredDoc);
+    /**
+     *
+     * @deprecated
+     * registerDoc is deprecated it will be removed in the next major release,
+     * use strapi.plugin('documentation').service('override').registerOverride instead
```

Comment:
Nice, I didn't know about this. Do you think we could go further and log a warning?